deploy/balena-k3s/README.md

```diff
@@ -1,27 +1,56 @@
 # Running the edge-endpoint via k3s on a balena device
 
 ## Setup
+First, install the balena CLI. You can find installation instructions for your platform at https://github.com/balena-io/balena-cli/blob/master/INSTALL.md
+
+Once installed, log in to your balena account:
+```bash
+balena login
+```
+
+## Deploying the edge-endpoint to a balena-managed device
+
+### Deploying to a CPU-only device
 Tested using an EC2 m6 instance with 64GB disk. Also known to work on a Raspberry Pi 5 with 8GB of RAM.
 
 From the root of `edge-endpoint`, run:
 ```bash
-balena login
-balena push <your-fleet>
+./deploy_balena.sh <my-fleet> cpu
+```
+
+This will build and push two services to the edge devices in your chosen fleet:
+1. A `server` service running a [k3s server](https://docs.k3s.io/architecture) that acts as the k3s cluster node
+2. A `bastion` service that provides access to the k3s cluster via kubectl commands and contains a copy of this repo at `/app/edge-endpoint`
+
+Now, we have our k3s single-node cluster built and running, but we have not started our edge deployment. To do this,
+see the [Configuration](#Configuration) section below (specifically, you will need to set the `RUN_EDGE_ENDPOINT` variable).
+
+### Deploying to an x86 device with an NVIDIA GPU
+Tested using an HP Victus Laptop with Intel Core i5 processor and NVIDIA GeForce RTX 3050 Laptop GPU.
+
+From the root of `edge-endpoint`, run:
+```bash
+./deploy_balena.sh <my-fleet> gpu
```

Comment:
I am also thinking about this as well. 

Right now this is probably fine because we only have either ARM CPU or x86 + CPU and we could just maintain two fleets. However, if we add Jetson into the mix (which also requires a different way to enable GPU support with a custom image for JetPack) it might become more complicated.

I am thinking maybe we can still keep the GPU container for CPU inference by modifying the launch script to not load Nvidia kernel modules and just idle instead (we can set device individual `INFERENCE_FLAVOR` in balena to control that or even modify the launch script to detect). This way we can keep all devices under the same fleet. Although this method will add more complexity so I am not sure if that is the right path we want to pursue.

Reply:
Fleets are supposed to be for one type of device in balena world are they not? I could try to see if theres a way to detect a-priori if we're running on a device with a compatible GPU

Reply:
Keeping fleets to a given device type makes a lot of sense. The deployment process itself would be cleaner if each fleet has a single type of device, even if the number of fleets we have gets messier.

In general, I'm a fan of pushing any work that we can into the deployment and build steps. I feel like we're already suffering from slow boots and processing limits

Reply:
The question here is "what is a 'type of device'". It's easy for that to explode, so maybe we want to spend some time thinking about that and pressure testing it.